src/ripple/protocol/Quality.h

```diff
@@ -234,6 +234,20 @@ class Quality
 
     friend
     bool
+    operator<= (Quality const& lhs, Quality const& rhs) noexcept
```

Comment:
Should we consider using `boost::totally_ordered`? See [`Rate`](https://github.com/ripple/rippled/blob/35fa20a110e3d43ffc1e9e664fc9017b6f2747ae/src/ripple/protocol/Rate.h#L38), [`XRPAmount`](https://github.com/ripple/rippled/blob/35fa20a110e3d43ffc1e9e664fc9017b6f2747ae/src/ripple/protocol/XRPAmount.h#L37) and [`IOUAmount`](https://github.com/ripple/rippled/blob/35fa20a110e3d43ffc1e9e664fc9017b6f2747ae/src/ripple/protocol/IOUAmount.h#L44).